docs/book/src/faq.md

```diff
@@ -0,0 +1,88 @@
+
+# Kubebuilder FAQ
+
+## How to undestand the domain field?
+First of all, we should get familiar with the GVK terminology, check [here](./cronjob-tutorial/gvks.md).
+
+The domain is for the group suffix, to explicitly show the resource group category.
+For example, if set `--domain=example.com`:
+```
+kubebuilder init --domain example.com --repo xxx --plugins=go/v4-alpha
+kubebuilder create api --group mygroup --version v1beta1 --kind Mykind
+```
+Then the result resource group will be `mygroup.example.com`.
+
+If domain field not set, the default value is `my.domain`.
+
+## How to validate API or set default value?
+We can use `webhook` to achieve it. See [webhook](./reference/webhook-overview.md)
+
+For example, we can create webhook with defaulting and validating logic by:
+```
+kubebuilder create webhook --group mygroup --version v1beta1 --kind Mykind --defaulting --programmatic-validation
+```
+Then the generated webhook code is like this:
+```go
+func (r *Mykind) Default() {
+	// TODO(user): fill in your defaulting logic.
+}
+
+func (r *Mykind) ValidateCreate() error {
+	// TODO(user): fill in your validation logic upon object creation.
+	return nil
+}
+
+func (r *Mykind) ValidateUpdate(old runtime.Object) error {
+	// TODO(user): fill in your validation logic upon object update.
+	return nil
+}
+
+func (r *Mykind) ValidateDelete() error {
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+```
+So it is easy to add API default and validate logic.
+
+## How to use `klog` or other loggers as the project logger?
```

Comment:
```suggestion
## I'd like to customize my project to use [klog][klog-lib] instead of the [zap][zap-lib] lib provided by controller-runtime. How to use `klog` or other loggers as the project logger?`

Reply:
Also, this one we might address in the scaffolds via https://github.com/kubernetes-sigs/kubebuilder/pull/3033 so we would not add here.